frontend/components/commons/sidebar/TaskSidebar.vue

```diff
@@ -81,6 +86,9 @@ export default {
       this.currentMetric = this.metricsByViewMode.includes(this.currentMetric)
         ? this.currentMetric
         : this.onShowSidebarInfo(false);
+      if (value === "labelling-rules" && this.isReferenceRecord) {
+        this.removeSimilarityFilter(value);
+      }
```

Comment:
Can we just emit an event with a view mode change and let the parent handle the action in that case?

I see here duplicated code that can be hard to maintain in the future if some of this logic changes.  What I propose is just:

```javascript
this.$emit("view-mode-changed", value);
```

and then, in the parent (which already knows about the record reference and vector stuff:
```javascript
onViewModeChanged(value) {
   if (if (value === "labelling-rules" && this.isReferenceRecord) {
        this.removeSimilarityFilter(value);
      }
}
```